src/ipc/ipc-common/IPCProxyModelProperty.h

```diff
@@ -139,10 +139,16 @@ class IPCProxyModelProperty : public facelift::ModelProperty<ModelDataType>
                     --last;
                 }
 
-                QList<ModelDataType> list;
-                m_proxy.ipc()->sendMethodCallWithReturn(requestMemberID, list, first, last);
-
-                Q_ASSERT(list.size() == (last - first + 1));
+                std::tuple<int, QList<ModelDataType>> requestResult;
+                m_proxy.ipc()->sendMethodCallWithReturn(requestMemberID, requestResult, first, last);
+
+                first = std::get<0>(requestResult);
+                auto &list = std::get<1>(requestResult);
+                // The server could change the number of returned items or even return an empty list
+                if (list.empty()) {
+                    return retval;
+                }
```

Comment:
@jacky309, @bitmouse, here we had qt_assert for empty list returned by the server, so we return an empty Model data.